omniscidb/QueryEngine/ExecutionKernel.cpp

```diff
@@ -150,6 +150,10 @@ void ExecutionKernel::run(Executor* executor,
   }
 }
 
+ExecutorDeviceType ExecutionKernel::getChosenDevice() const {
```

Comment:
Probably better `getChosenDeviceType`. Anyway, I think it's better to provide a `toString` method for all the logging.